### PR TITLE
BucketType method to locally store bucketType in variable

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Create BucketType() method to returns the type of bucket, allowing the use of a new API specifically designed for HNS buckets.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Added
3. Integration tests - Automated

### CODE REVIEW

File: internal/storage/gcs/bucket.go [hunk 1]
```diff
@@ -20,6 +20,15 @@ import (
 	"golang.org/x/net/context"
 )
 
+// BucketTypes represents different types of buckets.
```
Comment (ankitaluthra1) [inline internal/storage/gcs/bucket.go:23]: should be changed to "BucketType represent type of bucket like Hierarchical nonHierarchical as constants"

File: internal/storage/gcs/bucket.go [hunk 2]
```diff
@@ -20,6 +20,15 @@ import (
...
+type BucketTypes int
```
Comment (ankitaluthra1) [inline internal/storage/gcs/bucket.go:24]: please rename to BucketTypes as  at one time it can hold only one value, so should not be plural

**Review by ankitaluthra1 — DISMISSED** [review]
just few nits, please fix those, rest looks good to me

File: internal/storage/gcs/bucket.go [hunk 3]
```diff
@@ -20,6 +20,15 @@ import (
 	"golang.org/x/net/context"
 )
 
+// BucketType represent type of bucket like Hierarchical nonHierarchical as constants
```
Comment (raj-prince) [inline internal/storage/gcs/bucket.go:23]: nit: -> "BucketType represents different types of buckets like Hierarchical or NonHierarchical as constants."

**Review by raj-prince — DISMISSED** [review]

**Review by raj-prince — APPROVED** [review]

**Pull request merged**